deployment/tfhub/object-detection-url.ipynb

```diff
@@ -0,0 +1,607 @@
+      },
+      "source": [
+        "## 2. Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7SuugL_lOuaN"
+      },
+      "outputs": [],
+      "source": [
+        "import concurrent.futures\n",
+        "import os\n",
+        "import pandas as pd\n",
+        "import requests\n",
+        "import tempfile\n",
+        "import tensorflow as tf\n",
+        "import tensorflow_hub as hub\n",
+        "import time\n",
+        "import wget\n",
+        "\n",
+        "from bs4 import BeautifulSoup\n",
+        "from PIL import Image, ImageOps\n",
+        "from verta import Client\n",
+        "from verta.endpoint.autoscaling import Autoscaling\n",
+        "from verta.endpoint.autoscaling.metrics import CpuUtilizationTarget, MemoryUtilizationTarget, RequestsPerWorkerTarget\n",
+        "from verta.endpoint.resources import Resources\n",
+        "from verta.endpoint.update import DirectUpdateStrategy\n",
+        "from verta.environment import Python\n",
+        "from verta.registry import VertaModelBase, verify_io\n",
+        "from verta.utils import ModelAPI"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LB20pSYuXcCD"
+      },
+      "source": [
+        "## 3. Verta Set Up"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "hGwbnFGlXgJh"
+      },
+      "outputs": [],
+      "source": [
+        "os.environ['VERTA_HOST'] = ''\n",
+        "os.environ['VERTA_EMAIL'] = ''\n",
+        "os.environ['VERTA_DEV_KEY'] = ''"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "wVcbn1_3Xhtm",
+        "outputId": "f14f925f-7a48-4832-f95e-fc4de209514f"
+      },
+      "outputs": [],
+      "source": [
+        "client = Client(os.environ['VERTA_HOST'])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5bBHXvKnX_Je"
+      },
+      "source": [
+        "## 4. Model Class"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "o3YNlqVMEi6c"
+      },
+      "outputs": [],
+      "source": [
+        "class DetectObject(VertaModelBase):\n",
+        "    def __init__(self, artifacts=None):\n",
+        "        module_handle = 'https://tfhub.dev/google/openimages_v4/ssd/mobilenet_v2/1'\n",
+        "        self.detector = hub.load(module_handle).signatures['default']\n",
+        "    \n",
+        "    def handle_img(self, file_name, url, width=640, height=480):\n",
+        "        wget.download(url, file_name)\n",
+        "        img = Image.open(file_name)\n",
+        "        img = ImageOps.grayscale(img)\n",
+        "        img.thumbnail((width, height), Image.LANCZOS)\n",
```

Comment:
Do we need `img.save(path)` after this?

Reply:
No we don't. It was removed in last commits: [link](https://github.com/VertaAI/examples/blob/cbe12602bd9db63da7f2d88c22362aaa9ac89040/deployment/tfhub/object-detection-url.ipynb)

Reply:
Don't we need it, though? Otherwise

```python
img = ImageOps.grayscale(img)
img.thumbnail((width, height), Image.LANCZOS)
```

do not have any effect, I think, because we do not save this grayscale image.

Reply:
Just realized you your comment is probably about the [base64 version](https://github.com/VertaAI/examples/blob/cbe12602bd9db63da7f2d88c22362aaa9ac89040/deployment/tfhub/object-detection-base64.ipynb).

I'll check it out.

Reply:
Hmm understood. I'll check it out.

Reply:
You're right. The model was reading the original images without `image.save()`. I put it back, because
```
img = ImageOps.grayscale(img)
img.thumbnail((width, height), Image.LANCZOS)
```
are operations for performance optimization. :+1: